src/model/entities/device.ts

```diff
@@ -1,9 +1,95 @@
 import { StatusCode } from '../enums/status-code'
 
-export interface Device {
-  id: string
-  name: string
-  isConnected: boolean
-  statusCode: StatusCode
-  statusMessage: string
+export abstract class Device {
+  public type: string = 'Device'
+  public _id: string
+  private _name: string
+  private _isConnected: boolean
+  private _statusCode: StatusCode
+  private _statusMessage: string
+
+  constructor(id: string, name: string, isConnected: boolean, statusCode: StatusCode, statusMessage: string) {
+    this._id = id
+    this._name = name
+    this._isConnected = isConnected
+    this._statusCode = statusCode
+    this._statusMessage = statusMessage
+  }
+
+  public abstract connect(): void
+  
+  // Getters
```

Comment:
A comment like this explains _what_ the code is doing, not _why_ it is doing it.
If the code warrants a "what" comment, then the code isn't clear enough ;)